### PR TITLE
Adopt ALIRE compatible project files and build process

### PR DESCRIPTION
**Pull request by rod-chapman**
Prepare for contribution to ALIRE.

### CODE REVIEW

**Comment by Fabien-Chouteau** [comment]
Hi @rod-chapman, the `alire.toml` file should be checked-in.

**Comment by rod-chapman** [comment]
alire.toml added.  Not sure what it should contain yet!

**Comment by rod-chapman** [comment]
Note that I added project file parameter SPARKNACL_RUNTIME_MODE with values "full" and "zfp" to control selection of the runtime. This is in addition to all the normal parameters - I tried to follow the best practice guidance for those...

**Comment by Fabien-Chouteau** [comment]
> alire.toml added. Not sure what it should contain yet!

You can add `lincenses`, `tags`, `website`, and `long-description` fields: https://alire.ada.dev/docs/#catalog-format-specification
Also change the `description` to something like `SPARK 2014 re-implementation of the TweetNaCl crypto library`.

**Comment by Fabien-Chouteau** [comment]
In particular, you should have `spark` in the list of `tags` so that the project is identified as being in SPARK: https://alire.ada.dev/network

**Comment by Fabien-Chouteau** [comment]
> Note that I added project file parameter SPARKNACL_RUNTIME_MODE with values "full" and "zfp" to control selection of the runtime. This is in addition to all the normal parameters - I tried to follow the best practice guidance for those...

No problem.

You can declare `SPARKNACL_RUNTIME_MODE` in `alire.toml` so that Alire users know that this parameter exists:
```toml
[gpr-externals]
SPARKNACL_RUNTIME_MODE = ["full", "zfp"]
```

In future versions of Alire this info might be used to provide a warning or an error that the value should be set.

**Comment by rod-chapman** [comment]
OK... alire.toml updated - please check it looks OK.

**Comment by rod-chapman** [comment]
One question: SPARKNaCl uses the "Relaxed_Initialization" aspect and the 'Initialized attribute, both of which are only available in GNAT Community 2020 (or better) or GNAT Pro 20.x (??? is that right?)  So... can I specify in the alire.toml file that a particular version of GNAT is required?

**Comment by Fabien-Chouteau** [comment]
> OK... alire.toml updated - please check it looks OK.

Looks good to me!

> can I specify in the alire.toml file that a particular version of GNAT is required?
In theory you can with a `depends-on` entry:
```toml
[[depends-on]]
gnat = ">=2020"
```

**Comment by rod-chapman** [comment]
OK... added that dependency.  If you're happy, could you approve the Review please? I will merge after that...

**Review by Fabien-Chouteau — APPROVED** [review]

File: sparknacl.gpr [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+   for Create_Missing_Dirs use "True";
+   for Library_Dir use "lib";
+
+   type Library_Type_Type is ("relocatable", "static", "static-pic");
+   Library_Type : Library_Type_Type :=
+     external ("SPARKNACL_LIBRARY_TYPE", external ("LIBRARY_TYPE", "static"));
+   for Library_Kind use Library_Type;
+
+   type Enabled_Kind is ("enabled", "disabled");
+   Compile_Checks : Enabled_Kind := External ("SPARKNACL_COMPILE_CHECKS", "enabled");
```
Comment (Fabien-Chouteau) [inline sparknacl.gpr:17]: One more thing, for integration in Alire it is best to have the most performance by default, so disable all the checks. In particular for a SPARK project.

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]

File: sparknacl.gpr (hunk 1 above)
Comment (rod-chapman) [reply sparknacl.gpr:17]: OK... I will switch default build to no checks, no contracts and -O2 which offers the best size/performance tradeoff. (O3 is a tiny bit faster but results in *much* bigger code size owing to loop unrolling.)

**Pull request merged**